backend/common/utils/spatial.py

```diff
@@ -0,0 +1,158 @@
+import io
+import logging
+import os
+import shutil
+
+import numpy as np
+import pyvips
+from PIL import Image
+
+from backend.layers.thirdparty.s3_provider import S3Provider
+
+logger = logging.getLogger(__name__)
+
+
+class SpatialDataProcessor:
+    """
+    A class that processes spatial data and generates deep zoom assets.
+
+    Args:
+        s3_provider (S3Provider, optional): An instance of the S3Provider class. Defaults to None.
+
+    Attributes:
+        s3_provider (S3Provider): An instance of the S3Provider class.
+        deployment_stage (str): The deployment stage (e.g., "staging").
+        env (str): The environment based on the deployment stage.
+        bucket_name (str): The name of the S3 bucket.
+        asset_directory (str): The directory for the deep zoom assets.
+    """
+
+    def __init__(self, s3_provider: S3Provider = None):
+        self.s3_provider = s3_provider if s3_provider else S3Provider()
+        self.deployment_stage = os.getenv("DEPLOYMENT_STAGE", "staging")
+        # relying on dev bucket for rdev and test
+        self.env = "dev" if self.deployment_stage in ["rdev", "test"] else self.deployment_stage
+        self.bucket_name = f"spatial-deepzoom-{self.env}"
+        self.asset_directory = "spatial-deep-zoom"
+
+    def _calculate_aspect_ratio_crop(self, image_size):
+        """
+        Calculate the aspect ratio crop coordinates for an image.
+
+        Args:
+            image_size (int, int): The size of the image (width, height).
+
+        Returns:
+            tuple: The crop coordinates (left, upper, right, lower).
```

Comment:
what is the expected type int, float, or both?

Reply:
image_size will always be int because the image.shape will return (int, int)

the return value should be int, I've pushed an update to cast the returned tuple as int